src/test/java/org/beanpod/switchboard/controller/EncoderControllerTest.java

```diff
@@ -44,15 +47,19 @@
     //stubbed Objects
     static private DeviceEntity device1, device2;
     static private EncoderEntity encoder1, encoder2;
+    static private Set<ChannelEntity> setOfChannels;
     static private List<EncoderEntity> listOfEncoders;
 
     @BeforeAll
-    static void encoderFixture(){
+    static void encoderFixture() throws ParseException {
         //stubbing device and encoder objects
         device1 = DeviceFixture.getDevice1();
         device2 = DeviceFixture.getDevice2();
-        encoder1 = EncoderFixture.getEncoder1(device1);
-        encoder2 = EncoderFixture.getEncoder2(device2);
+
+        setOfChannels = ChannelFixture.getSetOfChannels();
+
+        encoder1 = EncoderFixture.getEncoder1(device1, setOfChannels);
+        encoder2 = EncoderFixture.getEncoder2(device2, setOfChannels);
         listOfEncoders = EncoderFixture.getListOfEncoder(encoder1, encoder2);
```

Comment:
Those fixtures should be initialized in the BeforeEach to avoid leaks in between tests. For example when you clear the encoder outputs

Reply:
fixed